dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -643,6 +643,56 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
       TypedDataset.create[Out](dropped)
     }
 
+  /**
+    * Drops a column from a Dataset
+    *
+    * @example
+    * {{{
+    *   case class X(i: Int, j: Int, k: Boolean)
+    *   case class Y(i: Int, k: Boolean)
+    *   val f: TypedDataset[X] = ???
+    *   val fNew: TypedDataset[Y] = f.drop[Y]('j)
+    * }}}
+    *
+    * @param column the column to drop as a symbol
+    * @param i0 TypeEncoder for output type U
+    * @param i1 LabelledGeneric derived for T
+    * @param i2 LabelledGeneric derived for U
+    * @param i3 diff from T to U
+    * @param i4 prof no fields have been added
+    * @param i5 the keys of removed fields
+    * @param i6 the one and only dropped column key
+    * @tparam U the output type
+    * @tparam TRep shapeless' record representation of T
+    * @tparam URep shapeless' record representation of U
+    * @tparam RemovedFields the removed fields from T to get to U
+    * @tparam RemovedKeys the keys of RemovedFields as an HList
+    *
+    * @see [[frameless.TypedDataset.dropApply#apply]]
+    */
+  def drop[U] = new dropApply[U]
```

Comment:
Looks kind of strange that the types are documented here, but they are not defined here ...

Reply:
Agreed that it looks strange but it's the same thing that I did https://github.com/typelevel/frameless/blob/4dbc2b3858eba1a67a952345aec990ff57855b53/dataset/src/main/scala/frameless/TypedDataset.scala#L669-L702

The reason for doing so is to get the nice scaladoc in IDEs etc on the method call site that the user will see. I don't have strong feelings either way but originally moved it due to the suggestion https://github.com/typelevel/frameless/pull/208#discussion_r151549415.

Whatever the decision, we should be consistent and update both this and `withColumn`.